### ISSUE TITLE
Random mouse behavior

### ISSUE THREAD

**Issue by pureconfig** [redox-os/redox#886]
Hi, discovered redox today. Looks like a cool project. I have a problem with the UI/mouse:

**Reproduction**: It happens consistently.

**Expected behavior**: Smooth mouse movement.

**Actual behavior**: The mouse cursor is slow, sometimes stops, and then speeds up again. Very hard to hit buttons in the UI because of this.

**Build information**: Latest available download as of today

**Comment by komawoyo** [comment]
more info on setup?

**Comment by ThomasPassmore** [comment]
Slow/laggy mouse cursor speed is linked to the size of the qemu window for me. Starting with a small 640x480 floating window (no matter the selected resolution at boot) results in a smooth cursor, scaling the window upwards in size creates this lag. On my machine it is not linked to the selected resolution when booting.


qemu version: `QEMU emulator version 2.6.1 (Debian 1:2.6.1+dfsg-0ubuntu5.3), Copyright (c) 2003-2008 Fabrice Bellard`
uname -a `Linux thomas-B85M-D3H 4.8.0-37-generic #39-Ubuntu SMP Thu Jan 26 02:27:07 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux`

Built using the default build instructions found [here](https://github.com/redox-os/redox#cloning-building-running) using up to date version of redox on master branch.

**Comment by zarevucky** [comment]
I've encountered similar behavior on HelenOS as a combination of processing being slow and QEMU sending data through PS/2 at the hardware mouse's rate (gaming mouse, much higher bandwidth than regular PS/2). Maybe it's a similar problem?

**Comment by pureconfig** [comment]
I'm on VMWare and VirtualBox, so definitely not QEMU related.

**Comment by komawoyo** [comment]
again, whats your setup/configuration?
Run: 
`VBoxManage showvminfo <vmware name>`

**Comment by polymetric1** [comment]
I know you mean 0.1.3, but 0.1.4 was just released a few minutes ago (as of now) so for future reference I'd recommend clarifying you're using 0.1.3 in the original post.

P.S. Try 0.1.4. it's probably not fixed but it won't hurt to check

**Comment by pureconfig** [comment]
@polymetric1 0.1.4 has the same problem. Moreover, 0.1.4 broke networking (I tried all the Intel cards, they all get stuck at dhcp)

**Comment by komawoyo** [comment]
@pureconfig Networking works for me

**Comment by pureformat** [comment]
@komawoyo It's just a bog standard virtual box - I didn't change anything, except trying all the intel cards.

**Comment by komawoyo** [comment]
@pureformat well whats your settings?

**Comment by pureformat** [comment]
@komawoyo 

```
Name:            Redox
Groups:          /
Guest OS:        Other/Unknown (64-bit)
Memory size:     1024MB
Page Fusion:     off
VRAM size:       16MB
CPU exec cap:    100%
HPET:            off
Chipset:         piix3
Firmware:        BIOS
Number of CPUs:  1
PAE:             on
Long Mode:       on
Triple Fault Reset: off
APIC:            on
X2APIC:          off
CPUID Portability Level: 0
CPUID overrides: None
Boot menu mode:  message and menu
Boot Device (1): Floppy
Boot Device (2): DVD
Boot Device (3): HardDisk
Boot Device (4): Not Assigned
ACPI:            on
IOAPIC:          on
BIOS APIC mode:  APIC
Time offset:     0ms
RTC:             local time
Hardw. virt.ext: on
Nested Paging:   on
Large Pages:     on
VT-x VPID:       on
VT-x unr. exec.: on
Paravirt. Provider: Default
Effective Paravirt. Provider: None
State:           powered off
Monitor count:   1
3D Acceleration: off
2D Video Acceleration: off
Teleporter Enabled: off
Teleporter Port: 0
Teleporter Address:
Teleporter Password:
Tracing Enabled: off
Allow Tracing to Access VM: off
Tracing Configuration:
Autostart Enabled: off
Autostart Delay: 0
Default Frontend:
Storage Controller Name (0):            IDE
Storage Controller Type (0):            PIIX4
Storage Controller Instance Number (0): 0
Storage Controller Max Port Count (0):  2
Storage Controller Port Count (0):      2
Storage Controller Bootable (0):        on
IDE (0, 0): Redox.vdi
IDE (1, 0): livedisk.iso
NIC 1:           MAC: 0800275AAF97, Attachment: NAT, Cable connected: on, Trace: off (file: none), Type: 82540EM, Reported speed: 0 Mbps, Boot priority: 0, Promisc Policy: deny, Bandwidth group: none
NIC 1 Settings:  MTU: 0, Socket (send: 64, receive: 64), TCP Window (send:64, receive: 64)
NIC 2:           disabled
NIC 3:           disabled
NIC 4:           disabled
NIC 5:           disabled
NIC 6:           disabled
NIC 7:           disabled
NIC 8:           disabled
Pointing Device: PS/2 Mouse
Keyboard Device: PS/2 Keyboard
UART 1:          disabled
UART 2:          disabled
UART 3:          disabled
UART 4:          disabled
LPT 1:           disabled
LPT 2:           disabled
Audio:           enabled (Driver: CoreAudio, Controller: AC97, Codec: STAC9700)
Clipboard Mode:  disabled
Drag and drop Mode: disabled
VRDE:            disabled
USB:             enabled
EHCI:            disabled
XHCI:            disabled

USB Device Filters:

<none>

Bandwidth groups:  <none>

Shared folders:  <none>

Video capturing:    not active
Capture screens:    0
Capture file:       Redox.webm
Capture dimensions: 1024x768
Capture rate:       512 kbps
Capture FPS:        25

Guest:

Configured memory balloon size:      0 MB

```

**Comment by komawoyo** [comment]
The main difference I see is the the following (my settings compared to yours). Additionally, I'm just booting by the livedisk.iso instead of the *.vdi - you should try removing the hard drive to test

```
VRAM size:       32MB
Paravirt. Provider: KVM
Effective Paravirt. Provider: KVM
```

**Comment by pureformat** [comment]
@komawoyo No, this happens on VMWare on Windows as well, on a different machine.

**Comment by komawoyo** [comment]
I'm on windows and that configuration you gave is from Virtualbox. Anyways, my recommendation is to stick to Virtualbox as I see more work being done for it. Stay away from VMware if you want to avoid problems

**Comment by pureformat** [comment]
@komawoyo You need to read more carefully. I said it happens on VMWare on Windows *as well*

So I have the same mouse problems on both VMWare (on Windows) and VirtualBox (on Mac)

**Comment by zarevucky** [comment]
You have dismissed my suggestion with "not using QEMU", but the point is that virtualized PS/2 may behave unrealistically because it's essentially dumb translation of what your hardware mouse is doing. In case you haven't already, try using a different mouse, or if your mouse has a settable frequency, set it to slowest possible. I do not suggest this as a solution, but to confirm or refute a hypothesis.

**Comment by zarevucky** [comment]
You can also try this: In VirtualBox, go into your VM's settings and change Pointing Device to "USB Tablet" instead of "PS/2 mouse". If my suspicion is correct, this will resolve the issue. VMWare probably has analogous settings, but I don't use it so I can't say where to find it.

**Comment by pureformat** [comment]
> In case you haven't already, try using a different mouse, or if your mouse has a settable frequency, set it to slowest possible.

I have tried it on VirtualBox on Mac, VMWare on Windows and my friend tried it on VirtualBox on Windows. Tried all relevant settings, different mice, as well as with/without acceleration.

I think it's time to admit the Redox mouse driver has a issues.

**Comment by le-jzr** [comment]
The question is not whether or not the mouse driver has issues. The question is how to investigate. I experienced the exact identical behavior, as I said, and tracked it down to the system (non-Redox) not being able to process the incoming interrupts fast enough. If this is a similar problem, then it's a combination of many influences that make this manifest. It might be that your PC has slower CPU than devs, or maybe all the mice you tried are gaming mice, and operating system is completely irrelevant. Unless @pureformat starts explicitly listing exact configurations, mice and CPU specs he has reproduced this with, we have nothing to work with.

**Comment by pureformat** [comment]
@le-jzr I posted my exact virtualbox spec above. WTF more do you need? I've tried several different PS2 mice, as well as USB mice with bios translation to ps2. All behave the same (weird).

**Comment by php5engineer** [comment]
Shot up a few videos to show what exactly we mean by erratic cursor movement
ASUS UN62V, i7-4600U, Redox 0.3.4 on LiveUSB, Canyon CNE-CMS1 mouse

[redox - horizontal vertical lines.m4v.zip](https://github.com/redox-os/redox/files/1752469/redox.-.horizontal.vertical.lines.m4v.zip)
[ubuntu - horizontal vertical lines.m4v.zip](https://github.com/redox-os/redox/files/1752468/ubuntu.-.horizontal.vertical.lines.m4v.zip)
[redox - moving a window.m4v.zip](https://github.com/redox-os/redox/files/1752476/redox.-.moving.a.window.m4v.zip)
